### PR TITLE
Subscription feature tests

### PR DESCRIPTION
**Pull request by miasma13**
Task/Issue URL:https://app.asana.com/0/1200019156869587/1207743313459997/f

**Description**:
Client side macOS Subscription feature tests.

**Steps to test this PR**:
1. Ensure normal app target builds correctly
2. Run full Unit tests suite.

**Definition of Done (Internal Only)**:

* [x] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)

### CODE REVIEW

**Review by aataraxiaa — COMMENTED** [review]
I went over this at a high-level, but will have another look tomorrow. The tests are well-structured, and nothing immediately stands out as a red flag. However, I left some comments.

File: UnitTests/Subscription/SubscriptionPagesUseSubscriptionFeatureTests.swift [hunk 1]
```diff
@@ -0,0 +1,1017 @@
...
+        subscriptionEnvironment = SubscriptionEnvironment(serviceEnvironment: .production,
+                                                          purchasePlatform: .appStore)
+        accountStorage = AccountKeychainStorageMock()
+        accessTokenStorage = SubscriptionTokenKeychainStorageMock()
+
+        entitlementsCache = UserDefaultsCache<[Entitlement]>(userDefaults: userDefaults,
+                                                             key: UserDefaultsCacheKey.subscriptionEntitlements,
+                                                             settings: UserDefaultsCacheSettings(defaultExpirationInterval: .minutes(20)))
+
+        // Real AccountManager
```
[the remark this answers is not in the document]
Comment (miasma13) [reply UnitTests/Subscription/SubscriptionPagesUseSubscriptionFeatureTests.swift:132]: The client side user script tests are more of an integration tests, they use "real" high level dependancies (managers and flows) and these are setup with mocked low-level services and storage.

**Review by aataraxiaa — APPROVED** [review]
Have looked at this more in-depth now, and have no concerns. The tests are validating the expected behavior and I don’t see any issues with the logic. Once any remaining comments are addressed, good to merge. Nice work!

**Review by aataraxiaa — APPROVED** [review]
Latest changes LGTM @miasma13. Just needs a main rebase/merge and then should be good to merge.

**Pull request merged**